### PR TITLE
ref(typing): stronger typing for utils/redis

### PR DESCRIPTION
**Pull request by anonrig**
Enables stronger typing for `sentry/utils/redis.py`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -628,6 +628,7 @@ disable_error_code = [
 # beginning: stronger typing
 [[tool.mypy.overrides]]
 module = [
+  "sentry.utils.redis",
   "src.sentry.tasks.on_demand_metrics",
   "src.sentry.relay.config.metric_extraction",
   "src.sentry.snuba.metrics.extraction",
```
Comment (asottile-sentry) [inline pyproject.toml:634]: hmmmm that would point at these being wrong -- do things break if these are corrected?

**Review by asottile-sentry — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (anonrig) [reply pyproject.toml:634]: Yes 👎 

```
src/sentry/snuba/metrics/extraction.py:568: error: Function is missing a return type annotation  [no-untyped-def]
src/sentry/snuba/metrics/extraction.py:572: error: Function is missing a return type annotation  [no-untyped-def]
src/sentry/snuba/metrics/extraction.py:576: error: Function is missing a type annotation  [no-untyped-def]
src/sentry/snuba/metrics/extraction.py:583: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
src/sentry/snuba/metrics/extraction.py:1199: error: Function is missing a return type annotation  [no-untyped-def]
src/sentry/snuba/metrics/extraction.py:1199: note: Use "-> None" if function does not return a value
src/sentry/snuba/metrics/extraction.py:1207: error: Function is missing a return type annotation  [no-untyped-def]
src/sentry/snuba/metrics/extraction.py:1281: error: Function is missing a return type annotation  [no-untyped-def]
src/sentry/relay/config/metric_extraction.py:570: error: Function is missing a return type annotation  [no-untyped-def]
src/sentry/relay/config/metric_extraction.py:575: error: Function is missing a return type annotation  [no-untyped-def]
src/sentry/relay/config/metric_extraction.py:586: error: Function is missing a return type annotation  [no-untyped-def]
src/sentry/relay/config/metric_extraction.py:1441: error: Function is missing a type annotation  [no-untyped-def]
src/sentry/relay/config/metric_extraction.py:1454: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
src/sentry/tasks/on_demand_metrics.py:186: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
src/sentry/tasks/on_demand_metrics.py:276: error: Function is missing a return type annotation  [no-untyped-def]
src/sentry/tasks/on_demand_metrics.py:314: error: Function is missing a return type annotation  [no-untyped-def]
src/sentry/tasks/on_demand_metrics.py:314: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
```

Comment (asottile-sentry) [reply pyproject.toml:634]: 😭

Comment (anonrig) [reply pyproject.toml:634]: I'll fix it on a different PR, don't worry!

**Pull request merged**